operator/internal/manifests/internal/config/loki-runtime-config.yaml

```
@@ -32,10 +32,10 @@ overrides:
     {{- if $l.PerStreamRateLimitBurst }}
     per_stream_rate_limit_burst: {{ $l.PerStreamRateLimitBurst }}MB
     {{- end }}
-    {{- if $l.DesiredRate }}
+    {{- with $l.DesiredRate }}
     shard_streams:
       enabled: true
-      desired_rate: {{ $l.DesiredRate }}
+      desired_rate: {{ . }}
```

I think this is either missing the `toBytes` or `MB` suffix:

```suggestion
      desired_rate: {{ . }}MB
```

---

Good catch!